boggle.js

```diff
@@ -20,7 +20,36 @@ function printBoggleBoard(boggleBoard) {
  * Shake a boggle board and fill it with letters.
  */
 function shake(boggleBoard) {
-  // This is your job. :)
+  // Create array of 6-sided dice
+  let dice = [
+    'AAEEGN',
+    'ELRTTY',
+    'AOOTTW',
+    'ABBJOO',
+    'EHRTVW',
+    'CIMOTU',
+    'DISTTY',
+    'EIOSST',
+    'DELRVY',
+    'ACHOPS',
+    'EIMNQU',
+    'EEINSU',
+    'EEGHNW',
+    'AFFKPS',
+    'HLNNRZ',
+    'DEILRX',
+  ];
+
+  for (let j = 0; j < boggleBoard.length; j++) {
+    for (let i = 0; i < boggleBoard[j].length; i++) {
+      // NOTE: I use splice below to prevent repetition of die on the board
+      let die = dice.splice(Math.floor(Math.random() * dice.length),1)[0];
```

Comment:
Deleting elements from the middle of an array is expensive. If an array has `n` element, deleting from the middle takes ~`n` operations. To delete every element from the array, one at a time, takes ~`n**2` operations.

Instead, consider shuffle the list of dice and then using a function like `randomElement` that takes in an array and returns a random element.

Iterating over the shuffled array guaranteed uniqueness.

Reply:
```javascript
let shuffledDice = shuffle(dice);

for (let i = 0; i < board.length; i++) {
  for (let j = 0; i < board[i].length; j++) {
    board[i][j] = randomElement(shuffledDice[i * board.length + j]);
  }
}
```